src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -57,98 +59,283 @@ public Class<CreateCollectionCommand> getCommandClass() {
+
+    // Validate the model and its vector dimension
+    return validateModelAndDimension(userConfig, providerConfig, userVectorDimension);
+  }
+
+  private PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig getAndValidateProviderConfig(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig) {
+    PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig =
+        embeddingProviderConfig.providers().get(userConfig.provider());
+    if (providerConfig == null || !providerConfig.enabled()) {
+      throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+          "Service provider '%s' is not supported", userConfig.provider());
+    }
+    return providerConfig;
+  }
+
+  // TODO: validate the 'secret_name' in the future
+  private void validateAuthentication(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
+      PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig) {
+    // Check if user authentication type is support
+    userConfig.vectorizeServiceAuthentication().type().stream()
+        .filter(type -> !providerConfig.supportedAuthentication().contains(type))
+        .findFirst()
+        .ifPresent(
+            type -> {
+              throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+                  "Authentication type '%s' is not supported", type);
+            });
+    // Check if 'secret_name' is provided if authentication type is 'SHARED_SECRET'
+    if (userConfig.vectorizeServiceAuthentication().type().contains("SHARED_SECRET")
+        && (userConfig.vectorizeServiceAuthentication().secretName() == null
+            || userConfig.vectorizeServiceAuthentication().secretName().isEmpty())) {
+      throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+          "'secret_name' must be provided for 'SHARED_SECRET' authentication type");
+    }
+  }
+
+  // TODO: what if user provide some useless parameters (i.e. not in config), just ignore?
+  private void validateUserParameters(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
+      PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig) {
+    if (providerConfig.parameters() == null) return;
+
+    providerConfig
+        .parameters()
+        .forEach(
+            expectedParamConfig -> {
+              Object userParamValue =
+                  userConfig.vectorizeServiceParameter().get(expectedParamConfig.name());
+              // if the user doesn't provide the parameters in config
+              if (userParamValue == null) {
+                // if the parameter is required, throw error
+                if (expectedParamConfig.required()) {
+                  throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+                      "Please provide required parameter '%s'.", expectedParamConfig.name());
```

Comment:
Can we rephrase it as `Required vectorize parameter %s for the provider %s missing`, don't use `Please` in the error message